src/Microsoft.TemplateSearch.TemplateDiscovery/Nuget/NugetPackProvider.cs

```diff
@@ -0,0 +1,119 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Net;
+using System.Net.Http;
+using Microsoft.TemplateSearch.TemplateDiscovery.PackProviders;
+using Newtonsoft.Json;
+
+namespace Microsoft.TemplateSearch.TemplateDiscovery.Nuget
+{
+    public class NugetPackProvider : IPackProvider
+    {
+        private static readonly string SearchUrlFormat = "https://api-v2v3search-0.nuget.org/query?q=template&skip={0}&take={1}&prerelease=true";
+
+        private readonly string _packageTempPath;
+        private readonly int _pageSize;
+        private readonly bool _runOnlyOnePage;
+
+        private static readonly string DownloadedPacksDir = "DownloadedPacks";
+
+        public NugetPackProvider(string packageTempBasePath, int pageSize, bool runOnlyOnePage)
+        {
+            _pageSize = pageSize;
+            _runOnlyOnePage = runOnlyOnePage;
+            _packageTempPath = Path.Combine(packageTempBasePath, DownloadedPacksDir);
+
+            if (Directory.Exists(_packageTempPath))
+            {
+                throw new Exception($"temp storage path for nuget packages already exists");
+            }
+            else
+            {
+                Directory.CreateDirectory(_packageTempPath);
```

Comment:
If possible, we should use the file system interfaces to make this testable

Reply:
This is outside the cone or context of any hive.

Reply:
`IPhysicalFileSystem` isn't bound to any hive